ceph/rados/core_workflows.py

```diff
@@ -1980,3 +1980,157 @@ def restart_daemon_services(self, daemon: str):
+        else:
+            for omap_obj in out_put:
+                obj_list.append(omap_obj["name"])
+        return obj_list
+
+    def get_object_key_list(self, osd_id, pg_id, object_name):
+        """
+        Method returns the key list of an object
+        Args:
+            osd: osd id number
+            pg_id: pg id number
+            object_name: object name
+
+        Returns: List of keys mapped to that object
+
+        """
+        cmd_base = f"cephadm shell --name osd.{osd_id} --"
+        cmd_get_obj_key = (
+            f"{cmd_base} ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-"
+            f"{osd_id} --pgid {pg_id} {object_name}  list-omap"
+        )
+        out_put = self.client.exec_command(sudo=True, cmd=cmd_get_obj_key)
+        key_list = list(filter(None, out_put[0].split("\n")))
+        return key_list
+
+    def rm_object_key(self, osd_id, pg_id, object_name, object_key):
+        """
+        Method removes the object key
+        Args:
+            osd_id: osd id number
+            pg_id: pg id number
+            object_name: object name
+            object_key: object key to remove that mapped to the object
+
+        Returns: True -> Key deleted False -> Key not deleted
+        """
+        cmd_base = f"cephadm shell --name osd.{osd_id} --"
+        cmd_rm_obj_key = (
+            f"{cmd_base} ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-"
+            f"{osd_id} --pgid {pg_id} {object_name}  rm-omap {object_key}"
+        )
+        try:
+            self.client.exec_command(sudo=True, cmd=cmd_rm_obj_key)
+        except Exception:
+            log.error(
+                f"{object_key} object key is not removed for the {object_name} object"
+            )
+            return False
+        log.info(f"{object_key} object key is removed for the {object_name} object")
+        return True
+
+    def get_inconsistent_pg_list(self, pool_name):
+        """
+        Method returns the inconsistent pg list
+        Args:
+            pool_name:  pool name
+
+        Returns: Inconsistent pg list
+        """
+        cmd_inconsist_pg = f"rados list-inconsistent-pg  {pool_name}"
+        inconsistent_pg_list = self.run_ceph_command(cmd_inconsist_pg)
+        if not inconsistent_pg_list:
+            log.info(f"The inconsistent pg list is empty in the pool {pool_name}")
+        return inconsistent_pg_list
+
+    def get_inconsistent_object_details(self, pg_id):
+        """
+        Method returns the inconsistent object list from a PG
+        Args:
+            pg_id: pg id
+
+        Returns: Return the inconsistent object list
+        """
+        cmd_inconsistent_object = f"rados  list-inconsistent-obj  {pg_id}"
+        object_details = self.run_ceph_command(cmd_inconsistent_object)
+        if not object_details:
+            log.info(f"The inconsistent objects list is empty in the PG {pg_id}")
+        return object_details
+
+    def chk_inconsistent_health(self):
```

Comment:
can the function be renamed to "check_inconsistent_health". As best practice lets not use short names instead use function names which reflect the logic being implemented in the function.

Reply:
Done.